idunn/geocoder/nlu_client.py

```diff
@@ -212,27 +226,57 @@ async def get_category():
 
         if category:
             return Intention(
+                type=IntentionType.CATEGORY,
                 filter={"category": category, "bbox": bbox},
                 description={"category": category, "place": place},
             )
 
         return Intention(
-            filter={"q": cat_query, "bbox": bbox}, description={"query": cat_query, "place": place}
+            type=IntentionType.BRAND,
+            filter={"q": cat_query, "bbox": bbox},
+            description={"query": cat_query, "place": place},
         )
 
     async def build_intention_category(self, cat_query, is_brand=False):
         category = await self.classify_category(cat_query, is_brand)
 
         if category:
-            return Intention(filter={"category": category}, description={"category": category})
+            return Intention(
+                type=IntentionType.CATEGORY,
+                filter={"category": category},
+                description={"category": category},
+            )
 
-        return Intention(filter={"q": cat_query}, description={"query": cat_query})
+        return Intention(
+            type=IntentionType.BRAND,
+            filter={"q": cat_query},
+            description={"query": cat_query},
+        )
 
     @classmethod
     def is_poi_request(cls, tags_list):
         """Check if a request is addressed to a POI"""
         return any(t.get("tag") in NLU_POI_TAGS for t in tags_list)
 
+    @classmethod
+    def is_street_request(cls, tags_list):
+        """
+        Check if a request only contains a street description.
+        >>> NLU_Helper.is_street_request([
+        ...     {"phrase": "3 rue des Rosiers", "tag": "street"},
+        ...     {"phrase": "Paris", "tag": "city"},
+        ... ])
+        True
+        >>> NLU_Helper.is_street_request([
+        ...     {"phrase": "restaurant", "tag": "cat"},
+        ...     {"phrase": "rue des Rosiers, Paris", "tag": "street"},
+        ... ])
+        False
+        """
+        contains_street_tag = any(t.get("tag") in NLU_STREET_TAGS for t in tags_list)
+        only_place_tags = all(t.get("tag") in NLU_STREET_TAGS + NLU_PLACE_TAGS for t in tags_list)
+        return contains_street_tag and only_place_tags
```

Comment:
I am still a bit reluctant to this implementation where any query like "Paris" (single tag `city`) will trigger an additional request to PJ, with little benefit.  But this can probably be discussed/adjusted separately.